.build/runbuild.ps1

```diff
@@ -200,4 +254,26 @@ function New-TemporaryDirectory {
 function Normalize-FileSystemSlashes([string]$path) {
     $sep = [System.IO.Path]::DirectorySeparatorChar
     return $($path -replace '/',$sep -replace '\\',$sep)
+}
+
+function DeleteCachedNuGetPackage([string]$packageId, [string]$version) {
+    # The user may have overridden the default .nuget package directory location using NUGET_PACKAGES env variable.
+    [string]$nugetCacheDir = $env:NUGET_PACKAGES
+    if (!(Test-Path $nugetCacheDir)) {
+        $nugetCacheDir = Join-Path $env:USERPROFILE "\.nuget\packages"
```

Comment:
This line fails on macOS, because `$env:USERPROFILE` is missing I think. You should be able to change this line to `$nugetCacheDir = "~/.nuget/packages"` and I tested that this should work on macOS, Linux, and Windows by running `Get-ChildItem "~/.nuget/packages"` which ran successfully on all 3 platforms. (You can also use backslashes, either way works.)

Reply:
It looks like the `Test-Path` condition will always pass if `$env:NUGET_PACKAGES` is empty because it interprets that as the current directory. So I think that replacing it with a check for null or empty would fix it. NuGet will create the cache directory if it doesn't already exist, so checking whether the directory exists is moot. We just need to point where it is configured (whether by the env variable or default).

Reply:
Yeah, and the other line will need to change in order to work xplat. You are right about that. On macOS and Linux, it is `$env:HOME`. So the ~ syntax is simpler.